packages/ai-ide-agents/src/browser/file-changeset-functions.ts

```diff
@@ -108,78 +106,134 @@ export class ReplaceContentInFileProvider implements ToolProvider {
 
-                    if (updatedContent !== fileContent) {
-                        let changeSet = ctx.session.changeSet;
-                        if (!changeSet) {
-                            changeSet = new ChangeSetImpl('Changes proposed by Coder');
-                            ctx.session.setChangeSet(changeSet);
-                        }
-
-                        changeSet.addOrReplaceElement(
-                            this.fileChangeFactory({
-                                uri: fileUri,
-                                type: 'modify',
-                                state: 'pending',
-                                targetState: updatedContent,
-                                changeSet,
-                                chatSessionId: ctx.session.id
-                            })
-                        );
-                    }
-                    return `Proposed replacements in file ${path}. The user will review and potentially apply the changes.`;
-                } catch (error) {
-                    console.info('Error processing replacements:', error.message);
-                    return JSON.stringify({ error: error.message });
+        return {
+            description: replacementDescription,
+            parameters: replacementParameters
+        };
+
+    }
+
+    async createChangesetFromToolCall(toolCallString: string, ctx: MutableChatRequestModel): Promise<string> {
+        try {
+            const { path, replacements } = JSON.parse(toolCallString) as { path: string, replacements: Replacement[] };
+            const fileUri = await this.workspaceFunctionScope.resolveRelativePath(path);
+            const fileContent = (await this.fileService.read(fileUri)).value.toString();
+
+            const { updatedContent, errors } = this.replacer.applyReplacements(fileContent, replacements);
+
+            if (errors.length > 0) {
+                return `Errors encountered: ${errors.join('; ')}`;
+            }
+
+            if (updatedContent !== fileContent) {
+                let changeSet = ctx.session.changeSet;
+                if (!changeSet) {
+                    changeSet = new ChangeSetImpl('Changes proposed by Coder');
+                    ctx.session.setChangeSet(changeSet);
                 }
+
+                changeSet.addOrReplaceElement(
+                    this.fileChangeFactory({
+                        uri: fileUri,
+                        type: 'modify',
+                        state: 'pending',
+                        targetState: updatedContent,
+                        changeSet,
+                        chatSessionId: ctx.session.id
+                    })
+                );
             }
+            return `Proposed replacements in file ${path}. The user will review and potentially apply the changes.`;
+        } catch (error) {
+            console.info('Error processing replacements:', error.message);
```

Comment:
Prefer `debug` if this isn't something that we'd expect ordinary users to be interested in regularly.